pkg/providers/tinkerbell/workers.go

```diff
@@ -43,6 +46,23 @@ func WorkersSpec(ctx context.Context, logger logr.Logger, client kubernetes.Clie
 		return nil, errors.Wrap(err, "generating tinkerbell workers yaml spec")
 	}
 
+	capiCluster, err := controller.GetCAPICluster(ctx, client, spec.Cluster)
+	if err != nil {
+		return nil, err
+	}
+	if capiCluster != nil {
+		rollingOverScale, err := anywherev1.IsRollingNotScale(ctx, client, spec.Cluster)
+		if err != nil {
+			return nil, errors.Wrap(err, "checking upgrade behavior")
+		}
+		if rollingOverScale {
+			workersYaml, err = omitTinkerbellMachineTemplate(workersYaml)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
```

Comment:
this logic is repeated int the CP
maybe that's a clue we should rethink how these 3 pieces interact

Reply:
will extract the logic to reconciler